test/test_util.h

```diff
@@ -1140,29 +1140,41 @@ struct NumericTraits<TestFoo>
 // Complex data type TestBar (with optimizations for fence-free warp-synchrony)
 //---------------------------------------------------------------------
 
+#define MAGIC 895245
 /**
  * TestBar complex data type
  */
 struct TestBar
 {
     long long       x;
     int             y;
+    int             magic;
 
     // Constructor
-    __host__ __device__ __forceinline__ TestBar() : x(0), y(0)
+    __host__ __device__ __forceinline__ TestBar() : x(0), y(0), magic(MAGIC)
     {}
 
     // Constructor
-    __host__ __device__ __forceinline__ TestBar(int b) : x(b), y(b)
+    __host__ __device__ __forceinline__ TestBar(int b) : x(b), y(b), magic(MAGIC)
     {}
 
     // Constructor
-    __host__ __device__ __forceinline__ TestBar(long long x, int y) : x(x), y(y)
+    __host__ __device__ __forceinline__ TestBar(long long x, int y) : x(x), y(y), magic(MAGIC)
     {}
 
+    // Assignment operator
+    __host__ __device__ __forceinline__ TestBar& operator =(const TestBar& that)
+    {
+        assert (magic == MAGIC);
```

Comment:
This should probably be something other than `assert` - in a release build, asserts may be turned off, and we want to catch cases where miscompiles in -O3 lead to missed assignments as well. Perhaps it should use the CUB error reporting machinery.